### PR TITLE
feat: open terminal

### PR DESCRIPTION
**Pull request by tomaszantas**
Description
---

- [x] Open terminal
- [x] Adjust Footer text (it always works on the `ctrl/cmd + T` shortcut

Motivation and Context
---

#29 

How Has This Been Tested?
---

Tested on Windows, Ubuntu and MacOS:

1. Start the application (development)
2. Press `Ctrl + T` (on MacOS use `Cmd + T`)
3. The Terminal should be open

MacOS:

https://user-images.githubusercontent.com/11715931/178327002-d4db9bee-8520-4d03-8a2f-e89781a79974.mp4

### CODE REVIEW

File: applications/launchpad/backend/src/commands/host.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+use std::{convert::TryFrom, path::PathBuf, process::Command, time::Duration};
+
+use log::*;
+use tauri::{api::path::home_dir, AppHandle, Wry};
+
+use crate::commands::AppState;
+
+#[tauri::command]
+pub async fn open_terminal(_app: AppHandle<Wry>, platform: String) -> Result<(), ()> {
+    let terminal_path = home_dir().unwrap().display().to_string();
```
Comment (tomaszantas) [inline applications/launchpad/backend/src/commands/host.rs]: @CjS77 @alex1307 I've made my attempt in Rust. Can you review pretty please all changes that I've done on backend, especially this line?

File: applications/launchpad/backend/src/commands/host.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+use std::{convert::TryFrom, path::PathBuf, process::Command, time::Duration};
+
+use log::*;
+use tauri::{api::path::home_dir, AppHandle, Wry};
+
+use crate::commands::AppState;
+
+#[tauri::command]
+pub async fn open_terminal(_app: AppHandle<Wry>, platform: String) -> Result<(), ()> {
```
Comment (alex1307) [inline applications/launchpad/backend/src/commands/host.rs]: I think at least an error message should be returned.
The response should be changed from  Result<(), ()> -to Result<(), String>

Comment (alex1307) [reply applications/launchpad/backend/src/commands/host.rs]: maybe better is  home_dir().ok_or()

File: applications/launchpad/backend/src/commands/host.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    let terminal_path = home_dir().unwrap().display().to_string();
+    if platform == "darwin" {
+        Command::new("open")
+            .args(["-a", "Terminal", &terminal_path])
+            .spawn()
+            .unwrap();
```
Comment (alex1307) [inline applications/launchpad/backend/src/commands/host.rs]: You should check the result and if there is an error then exit gracefuly.

File: applications/launchpad/backend/src/commands/host.rs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    } else if platform == "windows_nt" {
+        Command::new("powershell")
+            .args(["-command", "start", "powershell"])
+            .spawn()
+            .unwrap();
```
Comment (alex1307) [inline applications/launchpad/backend/src/commands/host.rs]: You should check the result and if there is an error then exit gracefuly.

File: applications/launchpad/backend/src/commands/host.rs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    } else if platform == "linux" {
+        Command::new("gnome-terminal")
+            .args([["--working-directory=", &terminal_path].join("")])
+            .spawn()
+            .unwrap();
```
Comment (alex1307) [inline applications/launchpad/backend/src/commands/host.rs]: You should check the result and if there is an error then exit gracefuly.

File: applications/launchpad/backend/src/commands/host.rs (hunk 3 above)
Comment (tomaszantas) [reply applications/launchpad/backend/src/commands/host.rs]: Tbh, I'm not familiar with Rust and not sure how I can do this. May I ask for a small code snippet?

**Review by tomaszantas — COMMENTED** [review]
Looks good, tested on my machine

**Review by corquaid — APPROVED** [review]
very nice!

**Pull request merged**